### ISSUE TITLE
iOS trouble

### ISSUE THREAD

**Issue by rossi256** [fechanique/cordova-plugin-fcm#135]
Hi,
I am having big trouble with this plugin on iOS

first of all i have many warnings on start of my app:
```
Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
FIRInstanceID AppDelegate proxy enabled, will swizzle app delegate remote notification handlers. To disable add "FirebaseAppDelegateProxyEnabled" to your Info.plist and set it to NO
Failed to fetch APNS token Error Domain=com.firebase.iid Code=1001 "(null)"
```

Receiving these messages even though i have 
`FirebaseAppDelegateProxyEnabled = NO` in googleService-info.plist.

And at each startup lots of warnings/errors:
```
1x Unable to connect to FCM. Error Domain=com.google.fcm Code=501 "(null)"
2x Failed to subscribe to topic Error Domain=com.google.fcm Code=5 "(null)"
4x Unable to connect to FCM. Error Domain=com.google.fcm Code=2001 "(null)"
```

This happens without any JS code added to the app!

status:
- app in background: receive test messages to single device on firebase, when clicking though i do not see the message in app (also not if i ran the code that is specified on main page)
- app in foreground: receive message.

**Comment by benlooi** [comment]
On Xcode 8/IOS 10, there is a lot of security stuff to prepare before it can work.
Have you gone to your apple developer account to create Provisioning profile?
You have to enable Push notification etc for the app ID.
If yes, then the next few things to take care of are:
Under Capabilities, turn on Push Notification, and Keychain sharing.

**Comment by rossi256** [comment]
yes i know as i wrote in #134  i have done all of that multiple times.
the provisioning profile works cause its a public app, created certificate request on mac, requested push notification dev-cert, added that to mac keychain, uploaded the p12 to firebase dev...

Only thing that i have not done is turn on Keychain sharing in Xcode, since it never anywhere suggested to do so.
just trying that and the errors all stay.

By the way i have now tried this one
https://github.com/arnesson/cordova-plugin-firebase
and there with all the rest the same setup, i have similar error/warning situation in xcode console but it seems to work
- foreground messages
- background message and when tapping background message, app opens and message popup appears.

**Comment by Eccenux** [comment]
@rossi256 Which iOS version are you testing? I'm trying to run my app on iOS 9.3 without luck. I can get the registration token, but don't receive any messages.

**Comment by rossi256** [comment]
iOS 10 - never tried on iOS 9.
did you try version 0.1.12? cause on latest version 0.1.13 kind of nothing works on iOS - at least iOS 10 for me it didn't work...

**Comment by Eccenux** [comment]
Oh. Then iOS 10 is not implemented (I checked the code). See #112 -> https://github.com/fechanique/cordova-plugin-fcm/issues/112#issuecomment-253186776

**Comment by rossi256** [comment]
those errors/warnings mentioned in my first post are only visible in the Xcode console... no errors in JS console...

**Comment by Eccenux** [comment]
If you have XCode then you could try adding changes required for iOS10 https://firebase.google.com/docs/notifications/ios/console-device#register_for_remote_notifications (seems like copy-paste mostly).

**Comment by rossi256** [comment]
sorry i would prefer someone who has some idea about this plugin to do that - i would burn too much time and don't know enough to really do this...

**Comment by Eccenux** [comment]
Hm... So I found a way to make it kind of work... Use below payload (message sent to FCM server) and it will work. Basically you cannot use the `notification` part (tested on both 1.1.2 and 1.1.5).

Some problems with this:

1. Without `notification` part Android notifications won't show up in the notifications list.
2. In iOS in version 1.1.2 made a vibration buzz when notification came. This is no longer available in 1.1.5.
3. In iOS for closed app (not in background) no notification is shown (same thing in 1.1.2 and 1.1.5, but in 1.1.2 it made a buzz).
4. In iOS for app in background notification is shown but with delay. Not a biggie. Seems like it is just how it works on iOS (probably due to battery concerns).

Kind of working payload:
```
{
	"data":{
		"title":"Notification title",
		"body":"Notification body",
		"message":"Notification message",
		"param1":"value1",
		"param2":"value2"
	},
	"to":"/topics/all",
	"priority":"high"
}
```

**Comment by Eccenux** [comment]
Oh. This is a PhoneGap Build configuration I use:
```xml
	<!-- Push Notifications - FCM -->
	<preference name="android-build-tool" value="gradle" />
	<plugin name="cordova-plugin-fcm" source="npm"  version="1.1.5" />
	<plugin name="cordova-plugin-fcm-config" spec="https://some-user:some-password@bitbucket.org/Eccenux/cordova-plugin-fcm-config.git" />
	<!--
		Otherwise will get:
		FirebaseInstanceID.framework/... was built for newer iOS version (7.0) than being linked (6.0)
	-->
	<preference name="deployment-target" value="7.0" />
```
Could probably set `deployment-target` to `8.0` as `isOS 7.0` is no longer supported in new PhoneGap/Cordova anyway.
